src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -228,6 +232,44 @@ public X509Certificate2 CreateAspNetCoreHttpsDevelopmentCertificate(DateTimeOffs
             return certificate;
         }
 
+        internal static bool CheckDeveloperCertificateKey(X509Certificate2 candidate)
+        {
+            // Tries to use the certificate key to validate it can't access it
+            try
+            {
+                // Search for the 'active' developer certificate, being very conservative about the possibility that there
+                // might be multiple repeated certificates and multiple HTTPS certificates that have been rolled over.
+                var devCert = ListCertificates(CertificatePurpose.HTTPS, StoreName.My,StoreLocation.CurrentUser, isValid:false)
```

Comment:
Why go through all this work when the candidate was already passed in?

Reply:
@Tratcher because the candidate cert has been "fiddled" with and I haven't been able to access the private key for it. (You can manipulate the instance in memory and have it "detached" from the key).

Reply:
How did that happen? Kestrel called the same CertificateManager APIs to get the cert.

Reply:
@Tratcher I'm not 100% sure, it can be that we do something slightly different or that I'm missing a "better pattern" for obtaining the key. This is the best that I've been able to do "empirically". 

Given that this runs on an exception code path and is for a development feature, I'm fine with the current code.

Reply:
I've figured this was a test artifact. The cert is being loaded in a way that makes the private key inaccessible. (I didn't change that, as it benefits my test case).

I've changed the cert in the affected test case for another cert that doesn't pretend to be the aspnet core dev cert, and used that "bad" original cert to add a test case for the error.

Reply:
> because the candidate cert has been "fiddled" with and I haven't been able to access the private key for it.

Isn't that exactly what the bug is though? The private key isn't available because of the protections in macOS Catalina? Maybe I'm misunderstanding it.

I do agree that this is not a performance-critical path.